toqito/matrices/cyclic_permutation.py

```diff
@@ -0,0 +1,61 @@
+"""Generate a cyclic permutation matrix."""
+
+import numpy as np
+
+
+def cyclic_permutation_matrix(n: int) -> np.ndarray:
+    r"""Create the cyclic permutation matrix for a given dimension :code:`n` :cite:`WikiCyclicPermutation`.
+
+    This function creates a cyclic permutation matrix which is a special type of square matrix
+    that represents a cyclic permutation of its rows.
+
+    The permutation can be written in cycle notation and two-line notation as:
+
+    .. math::
+        \begin{align*}
+        \begin{aligned}
+        &\begin{matrix}
+        (1 & 4 & 6 & 8 & 3 & 7)(2)(5)
+        \end{matrix}\\
+        &= \begin{pmatrix}
+        1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 \\
+        4 & 2 & 7 & 6 & 5 & 8 & 1 & 3
+        \end{pmatrix}\\
+        &= \begin{pmatrix}
+        1 & 4 & 6 & 8 & 3 & 7 & 2 & 5 \\
+        4 & 6 & 8 & 3 & 7 & 1 & 2 & 5
+        \end{pmatrix}
+        \end{aligned}
+        \end{align*}
```

Comment:
Everything under `math` is directly taken from Wikipedia. I am not too sure about this. We could have a user go to Wikipedia directly to look at this example there

Reply:
Fair enough. If the reference isn't enough, I'll remove it.

Reply:
Went ahead and removed for now.